spyder/plugins/editor/widgets/editor.py

```diff
@@ -1901,10 +1901,26 @@ def save(self, index=None, force=False, save_new_files=True):
 
         try:
             if self.format_on_save and finfo.editor.formatting_enabled:
-                # Autoformat document and then save
+                # Wait for document autoformat and then save
+
+                # Waiting for the autoformat to complete is needed
+                # when the file is going to be closed after saving.
+                # See spyder-ide/spyder#17836
+                format_eventloop = QEventLoop(None)
+                format_timeout = QTimer(self)

```

Comment:
What happens if the user tries to save two files in quick succession in the same editor stack? In that case, I think these two instances of QEventLoop and QTimer would be overwritten by the next save operation, which could cause a segfault.

To avoid that, perhaps they should be created at the CodeEditor level. Then, you could call them here like `finfo.editor.format_eventloop` and `finfo.editor.format_timeout`. Since it's not possible to save the same file twice or more times, then we would be safe.